core/src/stmt.rs

```diff
@@ -28,6 +28,11 @@ pub trait MongoStatement: Debug {
     fn execute(&mut self, _: &MongoConnection) -> Result<bool> {
         Err(Error::UnsupportedOperation("execute"))
     }
+    // Closes the cursor.
+    // Only MongoQuery supports this workflow. The other statements don't.
+    fn close_cursor(&mut self) -> Result<()> {
```

Comment:
Adding this to the trait seemed the most straightforward way to close the cursor when in the context of `SQLFreeStmt` since all we have access to at that point is the `mongo_statement` value which is typed as the trait `MongoStatement`.